pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx

```diff
@@ -0,0 +1,59 @@
+---
+description: A guide for migrating from v2 to v3 of the Hashboard CLI.
+---
+
+# V2 to V3 migration guide
+
+Version 3 of the Hashboard CLI launched in September 2024. Below is a guide to upgrading from Version 2 to Version 3.
+
+## **Major changes:**
+
+### **Creating and applying builds**
+
+There are no longer separate “Preview builds” and “Deploy builds”. Instead, you create a “build” once, which you can preview and then, if desired, apply directly. Applying a build is instantaneous.
+
+### **Builds are partial by default**
+
+When creating a build with the v2 CLI, resources that were previously deployed with code but no longer present were treated as deletions.
+
+In v3, Hashboard will only process changes for the files you’ve provided. For instance, you can choose to create a build that only includes your dashboards, even if you have yml files for other resources.
+
+```bash
+$ ls
+models		explorations			dashboards
+
+$ hb build dashboards/*  # Creates a build scoped just to your dashboards
+```
+
+Note that builds can still delete downstream content.  For example, if you update a model and remove one of its attributes, Explorations which depend on that attribute may be slated for removal, even if they are not in the build.
+
+Deployment of all code-controlled resources is still available by supplying a `--full_rebuild` flag:
```

Comment:
Would reword to `Rebuilding all code-controlled resources in the project is still ...`